tests/unittest/_torch/auto_deploy/unit/fused_moe/test_deepseek_v4_mxfp4_loader.py

```
@@ -0,0 +1,299 @@
+            layer_idx=3,
+            hidden_size=64,
+            intermediate_size=32,
+            num_experts=1,
+        )
+
+
+def test_loader_rejects_missing_expert_keys() -> None:
+    state, _ = _make_state_dict(num_experts=1, hidden_size=64, intermediate_size=32)
+    del state["layers.3.ffn.experts.0.w2.scale"]
+
+    with pytest.raises(DeepSeekV4MXFP4LoaderError, match="Missing.*w2\\.scale"):
```

Ruff (0.15.11)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)